app/serializers/site_category_serializer.rb

```diff
@@ -36,4 +36,19 @@ def include_allow_global_tags?
   def include_required_tag_groups?
     SiteSetting.tagging_enabled
   end
+
+  def notification_level
+    return if !scope || scope.anonymous?
+
+    CategoryUser.where(user_id: scope.user.id, category_id: object.id).pick(:notification_level) ||
```

Comment:
I think we have an N+1 query problem because `pick` triggers a query and this serializer is used in controllers to serialize array of records.

Reply:
Nice catch, this can be handled by `Category.preload_user_fields!`.